sopel/bot.py

```diff
@@ -665,7 +665,10 @@ def dispatch(self, pretrigger):
         event = pretrigger.event
         intent = pretrigger.tags.get('intent')
         nick = pretrigger.nick
-        is_echo_message = nick.lower() == self.nick.lower()
+        is_echo_message = (
+            nick.lower() == self.nick.lower() and
+            event in ["PRIVMSG", "NOTICE"]
```

Comment:
No need to wonder if the event will always be uppercase if you do this.

```suggestion
            event.upper() in ["PRIVMSG", "NOTICE"]
```

But there's plenty of code already assuming these will always be uppercase, so it's probably not worth doing only here. (FWIW, most of that code seems to use `'single quotes'`, but that's also likely not worth changing here.)